### PR TITLE
Use earthly to build kuberpult locally

### PR DESCRIPTION
**Pull request by ahmed-nour-fdc**
* Kuberpult can now be build using [earthly](https://github.com/earthly/earthly). Download the earthly binary and run `make kuberpult-earthly` to build the services in a containerised environment and run the services using the built images.

### CODE REVIEW

File: Earthfile [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+    RUN OS=Linux ARCH=$(uname -m) && \
+        SHA=$(cat buf_sha256.txt | grep "buf-${OS}-${ARCH}$" | cut -d ' ' -f1) && \
+        echo "${SHA}  ${BUF_BIN_PATH}/buf" | sha256sum -c
+    SAVE ARTIFACT go.mod
+    SAVE ARTIFACT go.sum
+    SAVE ARTIFACT buf_sha256.txt
+
+cd-service:
+    ARG UID=1000
+    ARG target=docker
```
Comment (sven-urbanski-freiheit-com) [inline Earthfile:23]: is it possible to define this only once? E.g. put it in the "deps" and depend on it?

Comment (ahmed-nour-fdc) [reply Earthfile:23]: Not yet supported, unfortunately.

File: pkg/path/Earthfile [hunk 2]
```diff
@@ -0,0 +1,9 @@
+VERSION 0.7
```
Comment (sven-urbanski-freiheit-com) [inline pkg/path/Earthfile:1]: that's a lot of files for everything in `pkg`.
Could we instead have one file `pkg/Earthfile` to take care of all of it?

File: services/cd-service/.arg [hunk 3]
```diff
@@ -0,0 +1 @@
+service=cd-service
```
Comment (sven-urbanski-freiheit-com) [inline services/cd-service/.arg:1]: Why do we need this?

File: services/cd-service/Earthfile [hunk 4]
```diff
@@ -0,0 +1,65 @@
+VERSION 0.7
+
+deps:
+    FROM ../../+deps
+    ARG service
+    ARG tag="deps"
+    COPY ../../pkg/api/+artifacts/* pkg/api/
```
Comment (sven-urbanski-freiheit-com) [inline services/cd-service/Earthfile:7]:
```suggestion
    COPY ../../pkg/*/+artifacts/* pkg/*/
```
?

File: services/cd-service/.arg (hunk 3 above)
Comment (ahmed-nour-fdc) [reply services/cd-service/.arg:1]: Removed

File: pkg/path/Earthfile (hunk 2 above)
Comment (ahmed-nour-fdc) [reply pkg/path/Earthfile:1]: It turned out to be easier than I thought. Now there is only one Earthfile in pkg

**Review by sven-urbanski-freiheit-com — APPROVED** [review]

**Pull request merged**